platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemReporter.java

```diff
@@ -57,30 +65,25 @@ public RuntimeException throwing(Action<ProblemSpec> spec) {
 
     public RuntimeException throwError(RuntimeException exception, Problem problem) {
         report(problem);
+        problems.put(exception, problem);
```

Comment:
❓ Why do we only associate the exception to the problem when we throw the exception via the Problem API? I think every time we report a problem that has an exception we should associate it, right?

Reply:
makes sense, yes.